### DISCUSSION TITLE
Ignore line changes in generated files

### DISCUSSION THREAD

**Discussion by magnattic** [community/community#12343] [Pull Requests]
## Motivation 
Right now, the line changes in a PR consider all files, including autogenerated ones. That makes it hard to find out how many **relevant** lines where changed, which would be useful to see how big the PR actually is.

One major use case for this is automatically labeling PRs by size, for example with https://github.com/CodelyTV/pr-size-labeler. This becomes much harder if the API also includes line changes from files that are already marked as autogenerated.

## Context
There is already the possibility to mark files as autogenerated through the `.gitattributes` file.

From the [github linguist docs](https://github.com/github/linguist/blob/master/docs/overrides.md#generated-code) (emphasis mine):

>Not all plain text files are true source files. Generated files like minified JavaScript and compiled CoffeeScript can be detected and excluded from language stats. **As an added bonus, unlike vendored and documentation files, these files are suppressed in diffs.** [...]
> 
>Use the linguist-generated attribute to mark or unmark paths as generated.
> `Api.elm linguist-generated`

Reading this I would expect those autogenerated files to also
- be excluded from the line count stats in a Github PR
- be excluded from the [Github API endpoint](https://docs.github.com/en/rest/reference/pulls#list-pull-requests-files) returning pull request files (or maybe marked in some way?)

## What actually happens

This PR shows 1,462 line changes, even though only 28 meaningful lines have changed. The rest is autogenerated.

![image](https://user-images.githubusercontent.com/7560166/156579287-c9c09556-d811-445c-8d27-5c807e03d8a7.png)

## What I would expect

Give us a way to see the **meaningful** line changes, both in the API and in the UI, by
- either excluding files marked as `linguist-generated` all-together in the PR and API
- or giving us another way to view these stats, maybe a second line count in the UI that ignores autogenerated files or another query in the API that gives you only the manually changed files.

**Comment by dynajoe** [comment]
Very painful for go `vendor` directory.

**Reply by wemgl** [reply]
Exactly why I'm here

**Comment by homeserve-lsaudon** [comment]
Yes, it would be cool if it ignored the lines count stats.

**Comment by wemgl** [comment]
This would be a great feature to have. Any updates?

**Comment by solussd** [comment]
This would be nice to have for giant XML files for UI design tools that might as well be binaries. Any AI bots over there at Github want to pick this up? :)

**Comment by cian-bayer** [comment]
Agreed this is very frustrating. Flutter/Dart has generated files - database files, mock files, database migrations - which can make a PR look like it has 10,000-20,000 lines of code changed.
